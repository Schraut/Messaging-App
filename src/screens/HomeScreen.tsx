import { StyleSheet, Text, View } from "react-native";
import { Theme, useTheme } from "../../resources/theme/theme";

export default function HomeScreen() {
  const theme = useTheme();
  const styles = stylesHandler(theme);

  return (
    <View style={styles.container}>
      <Text style={styles.text}>Home Screen</Text>
    </View>
  );
}

const stylesHandler = (theme: Theme) =>
  StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: theme.colors.background,
      alignItems: "center",
      justifyContent: "center",
    },
    text: {
      color: theme.colors.inverseBackground,
      fontFamily: theme.font.textPrimary.fontFamily,
      fontSize: theme.font.textPrimary.fontSize,
    },
  });
