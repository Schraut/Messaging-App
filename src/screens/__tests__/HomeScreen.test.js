import React from "react";
import renderer from "react-test-renderer";
import ThemeContent from "../../../resources/theme/theme";
import HomeScreen from "../HomeScreen";

describe("Default HomeScreen", () => {
  it("renders in light mode", () => {
    const tree = renderer
      .create(
        <ThemeContent>
          <HomeScreen />
        </ThemeContent>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders in dark mode", () => {
    const tree = renderer
      .create(
        <ThemeContent isDarkMode={true}>
          <HomeScreen />
        </ThemeContent>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
