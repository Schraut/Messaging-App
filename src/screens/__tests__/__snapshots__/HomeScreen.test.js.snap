// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default HomeScreen renders in dark mode 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontFamily": "System",
        "fontSize": 12,
      }
    }
  >
    Home Screen
  </Text>
</View>
`;

exports[`Default HomeScreen renders in light mode 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 12,
      }
    }
  >
    Home Screen
  </Text>
</View>
`;
